handout/architect/10_ways_to_misuse_angular.md

```diff
@@ -0,0 +1,439 @@
+
+
+### Subscribing to Data
+To access a data property use operators such as `map` or `pluck` to derive separate streams. This allows you to utilize the `async` pipe in the templates and removes the need for managing subscriptions in the component class.
+
+When accessing multiple properties subscribing to the base observable will likely be a better option.
+
+
+
+
+## 3. Using the Raw `Http` Service
+It is highly recommended to create a wrapper service for the Angular `Http` service. This allows you to create a layered architecture for consuming RESTful APIs. The wrapper service, let's call it `ApiService`, will be the base. This is where you can map each request to handle `res.json()`. It provides a centralized location for error handling, logging and attaching auth tokens.
+
+The next layer up will be domain specific services such as `UserService`, `PostsService`, `CommentsService`, etc. These services use the `ApiService` for making HTTP requests. This makes testing easier since now we only have to mock out the `ApiService` and not rely on Mocking HTTP responses. These services are also the perfect place to implement data transformations. Your app should not necessarily consume data as it is returned from the API. You should consider normalizing and reshaping it into a format that is better suited to managing state. [normalizr](https://github.com/paularmstrong/normalizr) is a great option for this task.
+
+You should avoid using using the `Http` service or `ApiService` directly in components.
+
+
+
+
+
+## 4. Not Considering the Injector Tree when Lazy Loading Modules
+Lazy loaded modules create their own branch on the Dependency Injection tree. This leads to some non-obvious effects:
+
+- Lazy loaded module providers (services) will not be available until the module has been loaded in.
+- These providers will not be available globally, but rather scoped to the lazy loaded module.
+
+This means that it's possible to have services that belong to a lazy loaded module, that are not accessible by the root module or any other eagerly loaded module of your application.
+
+For more detailed examples see [Lazy Loading and the Dependency Injection Tree](../modules/lazy-load-di.md), [Creating a Feature Module](../modules/feature-modules.md) and [Sharing the Same Dependency Injection Tree](../modules/shared-di-tree.md) sections.
+
+If you would like to provide a service as an application-level singleton use the `forRoot()` function. For providing services to feature modules you should always use `forChild()`.
+
+
+
+
+## 5. Confusing the Source of Truth Between Router and Application State
+The angular router allows us to define a client side routing structure and manages transitions between those routes. It does not however, manage or mutate your application state. You are responsible for updating the application state in response to navigation events.
```

Comment:
You certainly hit the main points of how you _should_ use the router, but the problem it's solving is still left a little unclear.

May be worth mentioning that it's "tempting" or "common" to find code trying to manually manage state as well as router events as two separate things rather than treating state changes such as "selecting an item" as a route transition to being with (i.e. triggered by `router.navigate()` vs. dispatching an action to update the store)